### PR TITLE
light / dark mode toggle feature 

### LINKED ISSUE #130: Dark mode
If this feature is approved by TPP, then we can prepare this for a future release.
I see the designs in https://www.figma.com/file/RRlG97YOIUFvYW1LaJ3Jh6/Updated-Features?type=design&node-id=29-3495&t=NtO5VG7ex6rLPqnx-0

### PR DESCRIPTION
**Pull request by kevinle623**
https://github.com/uoftblueprint/the-period-purse-2023/assets/74465053/8857a86e-74e7-4ee6-a1dc-e21206e4ede7

### CODE REVIEW

File: app/src/main/java/com/tpp/theperiodpurse/ui/setting/SettingScreenLayout.kt [hunk 1]
```diff
@@ -294,6 +345,7 @@ fun TrackingOptionButton(
         ) {
             Icon(
                 painter = icon,
+                tint = appViewModel.colorPalette.MainFontColor,
```
Comment (leowrites) [inline app/src/main/java/com/tpp/theperiodpurse/ui/setting/SettingScreenLayout.kt]: The asset for icon in here should be dark as well

File: app/src/main/java/com/tpp/theperiodpurse/ui/legal/TermsScreen.kt [hunk 2]
```diff
@@ -18,18 +18,19 @@ import androidx.navigation.compose.rememberNavController
 import com.tpp.theperiodpurse.R
 import com.tpp.theperiodpurse.ui.component.TopNavBar
 import com.tpp.theperiodpurse.ui.education.EducationBackground
+import com.tpp.theperiodpurse.ui.viewmodel.AppViewModel
 
 @SuppressLint("UnusedMaterialScaffoldPaddingParameter")
 @Composable
-fun TermsScreen(navController: NavHostController) {
+fun TermsScreen(navController: NavHostController, appViewModel: AppViewModel) {
```
Comment (leowrites) [inline app/src/main/java/com/tpp/theperiodpurse/ui/legal/TermsScreen.kt:25]: In dark mode the text in here cannot be seen, it's black text on dark background. I think composables in legal/CommonComponent needs to be updated and that should fix PrivacyScreen as well

File: app/src/main/java/com/tpp/theperiodpurse/ui/symptomlog/LogScreen.kt [hunk 3]
```diff
@@ -220,14 +224,14 @@ fun LogScreenLayout(
...
-        LogScreenTopBarContent(navController = navController, date = date)
+fun LogScreenTopBar(navController: NavController, date: LocalDate, appViewModel: AppViewModel) {
+    PopupTopBar(onClose = { navController.navigateUp() }, appViewModel = appViewModel) {
+        LogScreenTopBarContent(navController = navController, date = date, appViewModel = appViewModel)
     }
 }
 
 @Composable
-private fun LogScreenTopBarContent(navController: NavController, date: LocalDate) {
+private fun LogScreenTopBarContent(navController: NavController, date: LocalDate, appViewModel: AppViewModel) {
```
Comment (leowrites) [inline app/src/main/java/com/tpp/theperiodpurse/ui/symptomlog/LogScreen.kt:227]: The background color is too light compared to the design. It should be 0xFFA1A2A6

File: app/src/main/java/com/tpp/theperiodpurse/ui/symptomlog/ExercisePrompt.kt [hunk 4]
```diff
@@ -26,12 +26,13 @@ import com.tpp.theperiodpurse.R
...
 import com.tpp.theperiodpurse.data.model.LogSquare
 import com.tpp.theperiodpurse.ui.onboarding.EditNumberField
+import com.tpp.theperiodpurse.ui.viewmodel.AppViewModel
 import com.tpp.theperiodpurse.ui.viewmodel.LogViewModel
 import java.sql.Time
 
 @OptIn(ExperimentalComposeUiApi::class)
 @Composable
-fun ExercisePrompt(logViewModel: LogViewModel) {
+fun ExercisePrompt(logViewModel: LogViewModel, appViewModel: AppViewModel) {
```
Comment (leowrites) [inline app/src/main/java/com/tpp/theperiodpurse/ui/symptomlog/ExercisePrompt.kt:36]: The textbox cannot be seen in darkmode

File: app/src/main/java/com/tpp/theperiodpurse/ui/symptomlog/NotesPrompt.kt [hunk 5]
```diff
@@ -24,11 +24,12 @@ import androidx.compose.ui.text.input.VisualTransformation
 import com.tpp.theperiodpurse.data.model.LogPrompt
 import com.tpp.theperiodpurse.ui.theme.LogSelectedTextColor
 import com.tpp.theperiodpurse.ui.theme.MainFontColor
+import com.tpp.theperiodpurse.ui.viewmodel.AppViewModel
 import com.tpp.theperiodpurse.ui.viewmodel.LogViewModel
 
 @OptIn(ExperimentalComposeUiApi::class)
 @Composable
-fun NotesPrompt(logViewModel: LogViewModel) {
+fun NotesPrompt(logViewModel: LogViewModel, appViewModel: AppViewModel) {
```
Comment (leowrites) [inline app/src/main/java/com/tpp/theperiodpurse/ui/symptomlog/NotesPrompt.kt:34]: The textbox cannot be seen in darkmode

File: app/src/main/java/com/tpp/theperiodpurse/ui/symptomlog/SleepPrompt.kt [hunk 6]
```diff
@@ -21,12 +21,13 @@ import androidx.compose.ui.unit.dp
...
 import com.tpp.theperiodpurse.data.model.LogPrompt
 import com.tpp.theperiodpurse.ui.onboarding.EditNumberField
+import com.tpp.theperiodpurse.ui.viewmodel.AppViewModel
 import com.tpp.theperiodpurse.ui.viewmodel.LogViewModel
 import java.sql.Time
 
 @OptIn(ExperimentalComposeUiApi::class)
 @Composable
-fun SleepPrompt(logViewModel: LogViewModel) {
+fun SleepPrompt(logViewModel: LogViewModel, appViewModel: AppViewModel) {
```
Comment (leowrites) [inline app/src/main/java/com/tpp/theperiodpurse/ui/symptomlog/SleepPrompt.kt:32]: The textbox cannot be seen in darkmode

File: app/src/main/java/com/tpp/theperiodpurse/ui/setting/SettingScreenNavigation.kt [hunk 7]
```diff
@@ -149,59 +154,74 @@ fun SettingsScreen(
...
-                        context = context,
-                        viewModel = onboardViewModel,
-                        outController = outController,
-                        onboardUiState = onboardUiState,
-                        appUiState = appUiState,
-                        calUiState = calUiState,
-                        signout = signout,
-                    )
+                    if (appViewModel != null) {
+                        ResetDatabase(
```
Comment (leowrites) [inline app/src/main/java/com/tpp/theperiodpurse/ui/setting/SettingScreenNavigation.kt]: Can these just be one if statement?

File: app/src/main/java/com/tpp/theperiodpurse/ui/setting/SettingScreenNavigation.kt [hunk 8]
```diff
@@ -149,59 +154,74 @@ fun SettingsScreen(
...
-                )
+                if (appViewModel != null) {
+                    DeleteAccountScreen(
+                        appBar = SettingAppBar(
+                            currentScreen = "Delete Account",
+                            canNavigateBack = navController.previousBackStackEntry != null,
+                            navigateUp = { navController.navigateUp() },
+                            color = appViewModel.colorPalette.HeaderColor1,
+                            appViewModel = appViewModel
+                        ),
```
Comment (leowrites) [inline app/src/main/java/com/tpp/theperiodpurse/ui/setting/SettingScreenNavigation.kt]: If we are passing in appViewModel to SettingAppBar already, there is no need to pass in color

**Comment by leowrites** [comment]
In LoggingOptionsPopup, you should be able to update the color for the overlay when clicking on + in calendar

File: app/src/main/java/com/tpp/theperiodpurse/ui/component/CommonComponent.kt [hunk 9]
```diff
@@ -159,7 +173,7 @@ fun PopupTopBar(
         horizontalAlignment = Alignment.CenterHorizontally,
         modifier = modifier
             .fillMaxWidth()
-            .background(color = HeaderColor1)
+            .background(color = appViewModel.colorPalette.CalendarDayColor)
```
Comment (leowrites) [inline app/src/main/java/com/tpp/theperiodpurse/ui/component/CommonComponent.kt]: The color in here should be darker (similar to black)

File: app/src/main/java/com/tpp/theperiodpurse/ui/education/EducationHomeScreen.kt [hunk 10]
```diff
@@ -163,7 +167,7 @@ fun DYKCard(navController: NavHostController) {
 
 @OptIn(ExperimentalMaterialApi::class)
 @Composable
-fun PeriodProducts(navController: NavHostController, it: Product) {
+fun PeriodProducts(navController: NavHostController, it: Product, appViewModel: AppViewModel) {
```
Comment (leowrites) [inline app/src/main/java/com/tpp/theperiodpurse/ui/education/EducationHomeScreen.kt:170]: background color for the card should be transparent in darkmode

**Comment by kevinle623** [comment]
> In LoggingOptionsPopup, you should be able to update the color for the overlay when clicking on + in calendar

Nice catch, I added it

**Review by leowrites — CHANGES_REQUESTED** [review]

**Comment by leowrites** [comment]
Need verification on database and backup due to database change. Need to test migration

**Review by leowrites — APPROVED** [review]

**Pull request merged**